### PR TITLE
Rotate the avatar to align with the HMD while moving

### PR DESCRIPTION
**Pull request by hyperlogic**
MyAvatar: refactored updateFromHMDSensorMatrix() a bit by splitting it into several methods, because it was getting quite large and becoming hard to follow.

* beginStraighteningLean() - can be called when we would like to trigger a re-centering action.
* shouldBeginStraighteningLean() - contains some of the logic to decide if we should begin a re-centering action. For now it encapulates the capsule check.
* processStraighteningLean() - performs the actual re-centering calculation.

New code was added to MyAvatar::updateFromHMDSensorMatrix() to trigger re-centering when the avatar speed rises over a threshold.

Secondly the Rig::computeMotionAnimationState() state machine for animGraph added a state change hysteresis of 100ms.  This hysteresis should help smooth over two issues.

1. When the delta position is 0, because the physics timestep was not evaluated.
2. During re-centering due to desired motion, the avatar velocity can fluctuate causing undesired animation state fluctuation.

Thirdly, the third person camera boom is calculated a bit differently when wearing an HMD.  Basically the boom is behind the character's position/orientation i.e. hips, if the character's rotation changes the boom will swing behind the character.  After the addition of rotation comfort mode for avatar rotation, this is not that disorienting.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: e484a904a2da25dba7506411c0759480a2c4134f available at: https://gist.github.com/7071ad3d4bac5ed2a00a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6076.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6076.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1c1221597d7f0abb26223922ccb9534486cf35ad available at: https://gist.github.com/7071ad3d4bac5ed2a00a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6076.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6076.dmg

**Comment by howard-stearns** [comment]
One issue we noticed was that the camera boom could end up oddly placed, which Tony is going to look into for this PR.

Separately, we decided that this PR is just about adjusting the attitude (over a short time period) at the start of motion, but that the continuously adjusting the attitude during motion will be dealt with separately (not in this PR).

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3e7a6fd490cfbd1bf327d71d4d6a6b6d30389963 available at: https://gist.github.com/7071ad3d4bac5ed2a00a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6076.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6076.dmg
